### PR TITLE
feat(linter) eslint-plugin-unicorn - text-encoding-identifier-case

### PR DESCRIPTION
**Pull request by camc314**
part of #684

### CODE REVIEW

**Review by Boshen — DISMISSED** [review]

**Comment by Boshen** [comment]
How is `no_useless_escape` failing in CI 🤯

File: crates/oxc_ast/src/visit.rs [hunk 1]
```diff
@@ -1064,7 +1064,7 @@ pub trait Visit<'a>: Sized {
             }
             JSXAttributeValue::Element(elem) => self.visit_jsx_element(elem),
             JSXAttributeValue::Fragment(elem) => self.visit_jsx_fragment(elem),
-            JSXAttributeValue::StringLiteral(_) => {}
+            JSXAttributeValue::StringLiteral(lit) => self.visit_string_literal(lit),
```
Comment (camc314) [inline crates/oxc_ast/src/visit.rs:1067]: oh - before we didn't check jsx string literals, that's why its failing now.

`<foo attr='\d'></foo>` this feels like a case that should be in the fail list anyway?

Comment (Boshen) [reply crates/oxc_ast/src/visit.rs:1067]: https://facebook.github.io/jsx/#sec-jsx-string-characters

1.5.1 JSX String Characters
Historically, string characters within [JSXAttributeValue](https://facebook.github.io/jsx/#prod-JSXAttributeValue) and [JSXText](https://facebook.github.io/jsx/#prod-JSXText) are extended to allow the presence of [HTML character references](https://facebook.github.io/jsx/#sec-HTMLCharacterReference) to make copy-pasting between HTML and JSX easier, at the cost of not supporting \ [EscapeSequence](https://tc39.es/ecma262/#prod-annexB-EscapeSequence) of ECMAScript's [StringLiteral](https://tc39.es/ecma262/#prod-StringLiteral). We may revisit this decision in the future.

hmmm ... it's a valid test case. Let's create an issue for this.

**Review by Boshen — APPROVED** [review]

File: crates/oxc_ast/src/visit.rs (hunk 1 above)
Comment (Boshen) [reply crates/oxc_ast/src/visit.rs:1067]: https://github.com/web-infra-dev/oxc/issues/1061

**Pull request merged**